execution/executor/src/components/apply_chunk_output.rs

```diff
@@ -235,7 +235,7 @@ impl ApplyChunkOutput {
         let keeps_and_discards = transaction_outputs.iter().map(|t| t.status()).cloned();
         let retries = repeat(TransactionStatus::Retry).take(to_retry.len());
 
-        let status = if state_checkpoint_to_add.is_some() {
+        let status = if new_epoch_marker.is_some() || append_state_checkpoint_to_block.is_some() {
```

Comment:
Why do we need to check for `new_epoch_marker` ? Just checking for `append_state_checkpoint_to_block` should do right?

Reply:
`append_state_checkpoint_to_block` is Some() when block_gas_limit is turned on. We want to insert the state checkpoint execution result in `status` when there is reconfiguration, to ensure the invariant always holds. 

Another way to fix is change the mempool assert (check txn_output.len() == num_user_txns + 2 for block with reconfiguration), since block with reconfiguration does not have state checkpoint txn.

Reply:
I think the correct logic is
```
if append_state_checkpoint_to_block.is_some() {
    if reconfig {
      // add checkpoint and retry as the result
    } else {
      // add checkpoint and success as the result
    }
}
```

and line 232 should be removed?

Reply:
I will fix the mempool check instead :)